lib/octEncodeNormals.js

```diff
@@ -0,0 +1,175 @@
+                    writeBufferComponentType(source, newComponentType, octEncodedVector.x, byteOffset);
+                    writeBufferComponentType(source, newComponentType, octEncodedVector.y, byteOffset + newComponentByteLength);
+                    byteOffset += byteStride;
+                }
+                accessor.byteStride = byteStride;
+                accessor.componentType = newComponentType;
+                accessor.type = 'VEC2';
+                var minMax = findAccessorMinMax(gltf, accessor);
+                accessor.min = minMax.min;
+                accessor.max = minMax.max;
+                var materialId = primitive.material;
+                var material = materials[materialId];
+                var techniqueId = material.technique;
+                var attributeVariableName = patchedTechniques[techniqueId];
+                if (!defined(patchedTechniques[techniqueId])) {
+                    attributeVariableName = patchTechnique(gltf, techniqueId);
+                    patchedTechniques[techniqueId] = attributeVariableName;
+                }
+                var technique = techniques[techniqueId];
+                var programId = technique.program;
+                if (!defined(patchedPrograms[programId])) {
+                    patchProgram(gltf, programId, attributeVariableName);
+                    patchedPrograms[programId] = true;
+                }
+            }
+        }
+    }
+    uninterleaveAndPackBuffers(gltf);
+}
+
+function canOctEncode(accessor) {
+    return accessor.type === 'VEC3' && accessor.componentType === WebGLConstants.FLOAT;
+}
+
+function getNormalAccessors(gltf) {
+    var meshes = gltf.meshes;
+    var accessors = {};
+    for (var meshId in meshes) {
+        if (meshes.hasOwnProperty(meshId)) {
+            var mesh = meshes[meshId];
+            var primitives = mesh.primitives;
+            var primitivesLength = primitives.length;
+            for (var i = 0; i < primitivesLength; i++) {
+                var primitive = primitives[i];
+                var primitiveAttributes = primitive.attributes;
+                for (var semantic in primitiveAttributes) {
+                    if (primitiveAttributes.hasOwnProperty(semantic)) {
+                        if (semantic === 'NORMAL') {
+                            accessors[primitiveAttributes[semantic]] = primitive;
+                        }
+                    }
+                }
+            }
+        }
+    }
+    return accessors;
+}
+
+function patchTechnique(gltf, techniqueId) {
+    var techniques = gltf.techniques;
+    var technique = techniques[techniqueId];
+    var attributes = technique.attributes;
+    var parameters = technique.parameters;
+    var normalParameterName;
+    for (var parameterName in parameters) {
+        if (parameters.hasOwnProperty(parameterName)) {
+            var parameter = parameters[parameterName];
+            var semantic = parameter.semantic;
+            if (semantic === 'NORMAL') {
```

Comment:
Agreed. It's probably worth noting that I'm pretty sure the AGI model converter leaves the semantic as NORMAL when it applies oct-encoding so that might be something for @tfili to look at.